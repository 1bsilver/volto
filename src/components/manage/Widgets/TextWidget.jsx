/**
 * TextWidget component.
 * @module components/manage/Widgets/TextWidget
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Form, Input, Label } from 'semantic-ui-react';
import { map } from 'lodash';

/**
 * Component to display a text widget.
 * @function TextWidget
 * @returns {string} Markup of the component.
 */
const TextWidget = ({
  id,
  title,
  required,
  description,
  error,
  value,
  onChange,
}) => (
  <Form.Field required={required} error={error.length > 0}>
    <label htmlFor={`field-${id}`}>
      {title}
      {description && <span className="help">{description}</span>}
    </label>
    <Input
      id={`field-${id}`}
      name={id}
      value={value || ''}
      onChange={({ target }) =>
        onChange(id, target.value === '' ? undefined : target.value)}
    />
    {map(error, message => (
      <Label key={message} basic color="red" pointing>
        {message}
      </Label>
    ))}
  </Form.Field>
);

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
TextWidget.propTypes = {
  /**
   * Id of the field
   */
  id: PropTypes.string.isRequired,
  /**
   * Title of the field
   */
  title: PropTypes.string.isRequired,
  /**
   * Description of the field
   */
  description: PropTypes.string,
  /**
   * True if field is required
   */
  required: PropTypes.bool,
  /**
   * List of error messages
   */
  error: PropTypes.arrayOf(PropTypes.string),
  /**
   * Value of the field
   */
  value: PropTypes.string,
  /**
   * On change handler
   */
  onChange: PropTypes.func.isRequired,
};

/**
 * Default properties.
 * @property {Object} defaultProps Default properties.
 * @static
 */
TextWidget.defaultProps = {
  description: null,
  required: false,
  error: [],
  value: null,
};

export default TextWidget;
