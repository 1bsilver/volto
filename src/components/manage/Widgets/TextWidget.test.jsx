import React from 'react';
import renderer from 'react-test-renderer';
import TextWidget from './TextWidget';

test('renders a text widget component', () => {
  const component = renderer.create(
    <TextWidget id="my-field" title="My field" onChange={() => {}} />,
  );
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});
