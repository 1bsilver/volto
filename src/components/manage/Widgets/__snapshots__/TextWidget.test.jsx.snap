// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a text widget component 1`] = `
<div
  className="field"
>
  <label
    htmlFor="field-my-field"
  >
    My field
  </label>
  <div
    className="ui input"
  >
    <input
      disabled={undefined}
      id="field-my-field"
      name="my-field"
      onChange={[Function]}
      tabIndex={undefined}
      type="text"
      value=""
    />
  </div>
</div>
`;
